src/Authors.php

```diff
@@ -0,0 +1,81 @@
+<?php namespace Lean\Endpoints;
+
+use Leean\AbstractCollectionEndpoint;
+use Lean\Endpoints\Authors\Filter;
+
+/**
+ * Class that returns a collection of posts using dynamic arguments.
+ *
+ * @package Lean\Endpoints;
+ */
+class Authors extends AbstractCollectionEndpoint {
+
+	/**
+	 * Path of the new endpoint.
+	 *
+	 * @Override
+	 *
+	 * @since 0.1.0
+	 * @var String
+	 */
+	protected $endpoint = '/authors';
+
+	/**
+	 * Object that holds the current queried object on the site.
+	 *
+	 * @since 0.1.0
+	 * @var \WP_Query
+	 */
+	protected $query = null;
+
+	/**
+	 * Flag used to carry the value of the filter and avoid to call the function
+	 * N times inside of the loop.
+	 *
+	 * @since 0.1.0
+	 * @var bool
+	 */
+	protected $format_item = false;
+
+	/**
+	 * WP_Query Loop that has been triggered from the endpoint.
+	 *
+	 * @return array An array with the data associated with the request.
+	 */
+	protected function loop() {
+		$data = [];
+
+		$this->args['number'] = get_option( 'posts_per_page', 10 );
+
+		$this->query = new \WP_User_Query( $this->args );
+		if ( $this->query->total_users > 0 ) {
```

Comment:
@mitogh do you have a suggestion to limit this?

for example, making the role parameter required and allow only certain values?